### ISSUE TITLE
2. As a developer I need to redo the architecture of the code

### ISSUE THREAD

**Issue by tristancabel** [geomstats/geomstats#921]
probleme d'organisation de classes, besoin de design patterns (visiteur? )
cas d'exemple avec plusieurs metriques diff pour des varietes (manifold).  Combinaison de métriques en parallèle ou reliés .
cela doit aussi permettre une meilleure lisibilité, accessibilité pour des nouveaux contributeurs.

**Comment by nguigs** [comment]
Some obvious problems:
- Maybe a RiemannianManifold class could be usefull, to gather manifold and metric
- [GeneralLinear](https://github.com/geomstats/geomstats/blob/eeba7b7a652d45fc0053e35219c03627f2e8406f/geomstats/geometry/general_linear.py#L9) class does not inherit from LieGroup
- Related to both #920 and this issue: Hypersphere.Metric.norm is 10 times slower than gs.linalg.norm: 
```python
import geomstats.backend as gs
from geomstats.geometry.hypersphere import HypersphereMetric
from timeit import timeit
x = gs.array([1., 0., 0., 0.])
v = gs.array([0., 2., 1.5, .4])
metric = HypersphereMetric(3)
print(timeit(lambda: metric.norm(v, x), number=1000)) #  0.11182429000018601
print(timeit(lambda: gs.linalg.norm(v), number=1000)) # 0.008933541999795125
```
This is because EmbeddingMetric is used, and is supposed to allow to change the metric, but this feature is never used. Possible solution: have a specific class for the sphere with the ambient euclidean metric, using numpy operations directly, and a separate class relying on EmbeddingMetric to allow metric changes. Should the names be different
- Current [FiberBundle](https://github.com/geomstats/geomstats/blob/eeba7b7a652d45fc0053e35219c03627f2e8406f/geomstats/geometry/fiber_bundle.py#L9) class does not allow to set different metrics on the total space. Problematic for #902 
- What structure for co-metrics (see #883)
- Implement symmetric spaces

**Comment by tristancabel** [comment]
Tristan + nicolas will start to wrok on it.

**Comment by nguigs** [comment]
See #382 for discussion on this topic

**Comment by nguigs** [comment]
Some user request on the visualization module: #936

**Comment by nguigs** [comment]
This was addressed in #1006
